### PR TITLE
refactor: completion priority enum

### PR DESCRIPTION
**Pull request by LoZander**
Refators completion priority into an enum. Also simplifies parts of `TypeCompleter`, partly due to incompatibility with new priority enum. Also adds another completion type `TypeBuiltinPolyCompletion` for builtin polymorphic types.

Related to #8419

### CODE REVIEW

File: main/src/ca/uwaterloo/flix/api/lsp/provider/completion/KeywordCompleter.scala [hunk 1]
```diff
@@ -29,25 +29,25 @@ object KeywordCompleter {
...
+      Completion.KeywordCompletion("enum"     , Priority.High),
+      Completion.KeywordCompletion("type"     , Priority.High),
+      Completion.KeywordCompletion("instance" , Priority.High),
+      Completion.KeywordCompletion("mod"      , Priority.Low),
+      Completion.KeywordCompletion("eff"      , Priority.Lower),
+      Completion.KeywordCompletion("struct"   , Priority.Lower),
+      Completion.KeywordCompletion("sealed"   , Priority.Lowest),
+      Completion.KeywordCompletion("trait"    , Priority.Lowest),
+      Completion.KeywordCompletion("import"   , Priority.Lowest),
     )
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/api/lsp/provider/completion/KeywordCompleter.scala:43]: Much better. Now we are getting somewhere. 

Are the priorities here based on how frequently the keywords are?

**Review by magnus-madsen — APPROVED** [review]

**Comment by magnus-madsen** [comment]
I think this is a good step forward, so I will merge it.

File: main/src/ca/uwaterloo/flix/api/lsp/provider/completion/KeywordCompleter.scala (hunk 1 above)
Comment (LoZander) [reply main/src/ca/uwaterloo/flix/api/lsp/provider/completion/KeywordCompleter.scala:43]: OH, sorry. Yes, the priorities here are. I thought you had referenced the priorities of builtin types

Comment (LoZander) [reply main/src/ca/uwaterloo/flix/api/lsp/provider/completion/KeywordCompleter.scala:43]: but it's based on the priorities compared to all the other keywords in the list, not just the ones that they share a prefix with. That's the issue you pointed out in #8419. Currently, I'm working on that

**Pull request merged**